packages/web3/src/api/index.ts

```diff
@@ -82,6 +82,39 @@ export class NodeProvider implements INodeProvider {
   }
 }
 
+export interface RequestArguments {
+  path: string
+  method: string
+  params?: any
+}
+
+export class RemoteNodeProvider implements INodeProvider {
+  readonly wallets!: NodeApi<string>['wallets']
+  readonly infos!: NodeApi<string>['infos']
+  readonly blockflow!: NodeApi<string>['blockflow']
+  readonly addresses!: NodeApi<string>['addresses']
+  readonly transactions!: NodeApi<string>['transactions']
+  readonly contracts!: NodeApi<string>['contracts']
+  readonly multisig!: NodeApi<string>['multisig']
+  readonly utils!: NodeApi<string>['utils']
+  readonly miners!: NodeApi<string>['miners']
+  readonly events!: NodeApi<string>['events']
+
+  constructor(request: (request: RequestArguments) => Promise<any>) {
```

Comment:
Question: 
From the perspective of a client who uses `walletconnect`,  is the intention that the `walletconnect` provider will have a new method, e.g. `callNodeApi` that relays the `RequestArguments` to `walletconnect` client?